### PR TITLE
ThirdEye: make completeness check fully depend on dataset config

### PR DESCRIPTION
**Pull request by apucher**
CAUTION: contains database schema change, see create-schema.sql @ dataset_config_index

* select datasets for completeness check from dataset config index

* run completeness check on active datasets only

* remove datasetsToCheck configuration option

### CODE REVIEW

**Comment by puneetjaiswal** [comment]
Observed this NPE while a local build.. also having local test failures.
Could you please check an fix these:

`13:39:23.804 [pool-38-thread-3] ERROR c.l.t.c.c.DataCompletenessTaskRunner - Exception in data completeness checker task
java.lang.NullPointerException: null
	at com.linkedin.thirdeye.completeness.checker.DataCompletenessTaskUtils.getBucketNameToCountStarMap(DataCompletenessTaskUtils.java:211) ~[classes/:na]
	at com.linkedin.thirdeye.completeness.checker.DataCompletenessTaskUtils.getCountsForBucketsOfDataset(DataCompletenessTaskUtils.java:189) ~[classes/:na]
	at com.linkedin.thirdeye.completeness.checker.Wo4WAvgDataCompletenessAlgorithm.computeBaselineCountsIfNotPresent(Wo4WAvgDataCompletenessAlgorithm.java:68) ~[classes/:na]
	at com.linkedin.thirdeye.completeness.checker.DataCompletenessTaskRunner.executeCheckerTask(DataCompletenessTaskRunner.java:111) [classes/:na]
	at com.linkedin.thirdeye.completeness.checker.DataCompletenessTaskRunner.execute(DataCompletenessTaskRunner.java:46) [classes/:na]
	at com.linkedin.thirdeye.anomaly.task.TaskDriver.lambda$start$0(TaskDriver.java:75) [classes/:na]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[na:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[na:1.8.0_121]
	at java.lang.Thread.run(Thread.java:745) ~[na:1.8.0_121]
`
May be related to your previous checkin.

`Tests run: 93, Failures: 6, Errors: 0, Skipped: 0, Time elapsed: 118.154 sec <<< FAILURE! - in TestSuite
getBetweenClause(com.linkedin.thirdeye.client.pinot.PqlUtilsTest)  Time elapsed: 0.152 sec  <<< FAILURE!
java.lang.ExceptionInInitializerError: null
	at com.linkedin.thirdeye.client.DAORegistry.assertNotNull(DAORegistry.java:209)
	at com.linkedin.thirdeye.client.DAORegistry.getMetricConfigDAO(DAORegistry.java:150)
	at com.linkedin.thirdeye.client.pinot.PqlUtils.<clinit>(PqlUtils.java:41)
	at com.linkedin.thirdeye.client.pinot.PqlUtilsTest.getBetweenClause(PqlUtilsTest.java:30)

getBetweenClause(com.linkedin.thirdeye.client.pinot.PqlUtilsTest)  Time elapsed: 0.001 sec  <<< FAILURE!
java.lang.NoClassDefFoundError: Could not initialize class com.linkedin.thirdeye.client.pinot.PqlUtils
	at com.linkedin.thirdeye.client.pinot.PqlUtilsTest.getBetweenClause(PqlUtilsTest.java:30)

getBetweenClause(com.linkedin.thirdeye.client.pinot.PqlUtilsTest)  Time elapsed: 0.002 sec  <<< FAILURE!
java.lang.NoClassDefFoundError: Could not initialize class com.linkedin.thirdeye.client.pinot.PqlUtils
	at com.linkedin.thirdeye.client.pinot.PqlUtilsTest.getBetweenClause(PqlUtilsTest.java:30)

getBetweenClause(com.linkedin.thirdeye.client.pinot.PqlUtilsTest)  Time elapsed: 0 sec  <<< FAILURE!
java.lang.NoClassDefFoundError: Could not initialize class com.linkedin.thirdeye.client.pinot.PqlUtils
	at com.linkedin.thirdeye.client.pinot.PqlUtilsTest.getBetweenClause(PqlUtilsTest.java:30)

getBetweenClause(com.linkedin.thirdeye.client.pinot.PqlUtilsTest)  Time elapsed: 0 sec  <<< FAILURE!
java.lang.NoClassDefFoundError: Could not initialize class com.linkedin.thirdeye.client.pinot.PqlUtils
	at com.linkedin.thirdeye.client.pinot.PqlUtilsTest.getBetweenClause(PqlUtilsTest.java:30)

getBetweenClause(com.linkedin.thirdeye.client.pinot.PqlUtilsTest)  Time elapsed: 0.001 sec  <<< FAILURE!
java.lang.NoClassDefFoundError: Could not initialize class com.linkedin.thirdeye.client.pinot.PqlUtils
	at com.linkedin.thirdeye.client.pinot.PqlUtilsTest.getBetweenClause(PqlUtilsTest.java:30)


Results :

Failed tests: 
com.linkedin.thirdeye.client.pinot.PqlUtilsTest.getBetweenClause(com.linkedin.thirdeye.client.pinot.PqlUtilsTest)
  Run 1: PqlUtilsTest.getBetweenClause:30 ExceptionInInitializer
  Run 2: PqlUtilsTest.getBetweenClause:30 NoClassDefFound Could not initialize class co...
  Run 3: PqlUtilsTest.getBetweenClause:30 NoClassDefFound Could not initialize class co...
  Run 4: PqlUtilsTest.getBetweenClause:30 NoClassDefFound Could not initialize class co...
  Run 5: PqlUtilsTest.getBetweenClause:30 NoClassDefFound Could not initialize class co...
  Run 6: PqlUtilsTest.getBetweenClause:30 NoClassDefFound Could not initialize class co...

`

**Comment by puneetjaiswal** [comment]
Some tests failed while building locally, may be related to your previous checkin.
[test.results.txt](https://github.com/linkedin/pinot/files/759101/test.results.txt)

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/completeness/checker/DataCompletenessScheduler.java [hunk 1]
```diff
@@ -1,46 +1,29 @@
...
   public void start() {
-
     LOG.info("Starting data completeness checker service");
 
     dataCompletenessJobContext = new DataCompletenessJobContext();
-    dataCompletenessJobContext.setDatasetsToCheck(datasetsToCheck);
     dataCompletenessJobRunner = new DataCompletenessJobRunner(dataCompletenessJobContext);
 
-    scheduledExecutorService.scheduleWithFixedDelay(dataCompletenessJobRunner, 0, 15, TimeUnit.MINUTES);
+    scheduledExecutorService.scheduleWithFixedDelay(dataCompletenessJobRunner, 0, 5, TimeUnit.MINUTES);
```
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/completeness/checker/DataCompletenessScheduler.java]: why 5 minutes and not 15 ? 
15 looks reasonable.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datalayer/entity/DatasetConfigIndex.java [hunk 2]
```diff
@@ -3,6 +3,7 @@
 public class DatasetConfigIndex extends AbstractIndexEntity {
   String dataset;
   boolean active;
+  boolean requiresCompletenessCheck;
```
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datalayer/entity/DatasetConfigIndex.java]: I would have named it "completenessCheckEnabled"

Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datalayer/entity/DatasetConfigIndex.java]: it's stronger than just enabling completeness check -- the detector will require a check to be performed before applying anomaly detection

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/completeness/checker/DataCompletenessScheduler.java (hunk 1 above)
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/completeness/checker/DataCompletenessScheduler.java]: This shouldn't even be here. A config file would be a better place. Suggestions?

**Comment by apucher** [comment]
Test failures are heisen-bugs caused by incomplete state reset between tests. Fix here: https://github.com/linkedin/pinot/pull/1040

**Comment by apucher** [comment]
rebased on master after merging #1040

**Pull request merged**